scripts/release-checks/dependency-ranges/latest-versions-check.mts

```diff
@@ -6,30 +6,31 @@
  * found in the LICENSE file at https://angular.dev/license
  */
 
-import { readFile } from 'node:fs/promises';
+import { createRequire } from 'node:module';
+import { fileURLToPath } from 'node:url';
 import semver from 'semver';
 
-export async function checkSchematicsAngularLatestVersion(
-  newVersion: semver.SemVer,
-): Promise<string[]> {
-  const { dependencies } = JSON.parse(
-    await readFile('./packages/schematics/angular/utility/latest-versions/package.json', 'utf-8'),
-  );
+export function checkSchematicsAngularLatestVersion(newVersion: semver.SemVer): string[] {
+  // Root of the Angular CLI project.
+  const root = fileURLToPath(new URL('../../../', import.meta.url));
+  const rootRequire = createRequire(root);
+  const { latestVersions } = rootRequire('./dist/@schematics/angular/utility/latest-versions.js');
```

Comment:
Is `dist/` guaranteed to be existent here? Seems like an assumption that is potentially brittle. Also the `createRequire` is not super great :(

Reply:
Yes, the NPM packages are always built in the `dist` https://github.com/angular/angular-cli/blob/c0d20e05c3470ff69576227d03cb913b1ab1337f/scripts/build-packages-dist.mts#L37


Regarding `createRequire`, I agree that using a `RegExp` could work. However, I lean toward this approach since it minimizes the risk of false positives or negatives.

Reply:
Ideally we'd somehow pass the packages and their artifact paths to the pre=release checks, but that's not possible right now, so LGTM